tests/gis_tests/geoapp/test_regress.py

```diff
@@ -87,3 +88,12 @@ def test_boolean_conversion(self):
         # verify values
         self.assertIs(val1, True)
         self.assertIs(val2, False)
+
+    def test_filter_on_distance_annotation(self):
+        "Geometric annotations work properly. See #27014."
```

Comment:
Try to describe the issue in the docstring (if any more explanation is needed) so we don't need the ticket reference.